include/System/MarioGamePad.hpp

```diff
@@ -1,20 +1,306 @@
+				stickScaling = 0.0f;
+			} else {
+				// TODO: CLBCalcRatio
+				// stickScaling =
+				stickScaling = CLBCalcRatio<s16>(0x28, 0x3c, _unk2);
+			}
+		}
+
+		if (_unk3) {
+			this->mCompSPos[0].x = stickScaling * this->mMainStick.mPosX;
+			this->mCompSPos[0].y = stickScaling * this->mMainStick.mPosY;
+		} else {
+			this->mCompSPos[0].x = this->mMainStick.mPosX;
+			this->mCompSPos[0].y = this->mMainStick.mPosY;
+		}
+
+		this->mCompSPos[1].x = (f32)this->mButton.mAnalogL;
+		this->mCompSPos[1].y = (f32)this->mButton.mAnalogR;
+
+		this->updateMeaning(A, MEANING_0x80, prevMeaning);
+
+		if ((this->mFlags & 0x4) != 0) {
+			this->updateMeaning(B, MEANING_0x800, prevMeaning);
+		} else {
+			if ((this->mFlags & 0x20) == 0) {
+				this->updateMeaning(B, MEANING_0x100, prevMeaning);
+			}
+		}
+
+		this->updateMeaning(R, MEANING_0x400, prevMeaning);
+		this->updateMeaning(Z, MEANING_0x1000, prevMeaning);
+		this->updateMeaning(L, MEANING_0x2000, prevMeaning);
+
+		this->mCompSPos[3].x = this->mSubStick.mPosX;
+		this->mCompSPos[3].y = this->mSubStick.mPosY;
+
+		this->updateMeaning(Y, MEANING_0x4000, prevMeaning);
+		this->updateMeaning(L, MEANING_0x8000, prevMeaning);
+
+		if (((this->_DE & 0x1) != 0)
+		    || (((this->_DC & 1) != 0
+		         && ((prevMeaning & MEANING_0x200) != 0)))) {
+			this->mMeaning |= MEANING_0x200;
+		}
+
+		this->updateMeaning(X, MEANING_0x200000, prevMeaning);
+	}
+
+	// Fabricated
+	inline void updateMeaning0x1(u32 prevMeaning)
+	{
+		// Some kind of 2d menu navigation?
+		if (mButton.mRepeat & (MAINSTICK_UP | DPAD_UP)) {
+			mMeaning |= MEANING_0x2;
+		}
+		if (mButton.mRepeat & (MAINSTICK_DOWN | DPAD_DOWN)) {
+			mMeaning |= MEANING_0x4;
+		}
+		if (mButton.mRepeat & (MAINSTICK_LEFT | DPAD_LEFT)) {
+			mMeaning |= MEANING_0x8;
+		}
+		if (mButton.mRepeat & (MAINSTICK_RIGHT | DPAD_RIGHT)) {
+			mMeaning |= MEANING_0x10;
+		}
+		this->updateMeaning(A, MEANING_0x20, prevMeaning);
+		this->updateMeaning(B, MEANING_0x40, prevMeaning);
+	}
+
+	// Fabricated
+	inline void updateMeaning0x8(u32 prevMeaning)
+	{
+		// This is definitely some kind of menu navigation
+		if (mButton.mRepeat & (MAINSTICK_UP | DPAD_UP)) {
+			mMeaning |= MEANING_0x80000;
+		}
+		if (mButton.mRepeat & (MAINSTICK_DOWN | DPAD_DOWN)) {
+			mMeaning |= MEANING_0x100000;
+		}
+		this->updateMeaning(A, MEANING_0x20000, prevMeaning);
+		this->updateMeaning(B, MEANING_0x40000, prevMeaning);
+		this->mCompSPos[1].x = (f32)(this->mButton.mAnalogL);
+		this->mCompSPos[1].y = (f32)(this->mButton.mAnalogR);
+		this->updateMeaning(R, MEANING_0x400, prevMeaning);
+		this->updateMeaning(Z, MEANING_0x1000, prevMeaning);
+		this->updateMeaning(L, MEANING_0x2000, prevMeaning);
+	}
+	// Fabricated
+	inline void updateMeaning0x80(u32 prevMeaning)
+	{
+		this->updateMeaning(A, MEANING_0x20, prevMeaning);
+		this->updateMeaning(B, MEANING_0x40, prevMeaning);
+		this->mCompSPos[2].x = this->mMainStick.mPosX;
+		this->mCompSPos[2].y = this->mMainStick.mPosY;
+	}
```

Comment:
how did you decide on these particular inlines? does it help matching it except for stack size?
because it doesn't look typical to me for them to put so much code into headers, tbh =\
I can understand `updateMeaning(EButtons button, PadMeanings padButton, u32 prevMeaning)`, but the others look way too unrealistic

Reply:
After giving it some thought, yes, lets flatten it back, as that makes a lot more sense in the context of how the rest of the game was written (except for the `updateMeaning(EButtons button, PadMeanings padButton, u32 prevMeaning)` inline which makes a lot of sense)